block.rb

```diff
@@ -1,16 +1,8 @@
-def factorial
-    #_______________________
-    yield
-end
-
-n = gets.to_i
-factorial do 
-  value =1
-  while ( n > 1)
-  value *= n
-  n-=1
-  end
-
-    puts value
+def factorial   
+  yield   
 end
 
+n = gets.to_i   
+factorial do    
+  puts (1..n).inject(:*)   
```

Comment:
đoạn `(1..n).inject(:*) ` bỏ lên hàm factorial em, ở đây mình chỉ in result thôi